### PR TITLE
docs: add coverage

### LINKED ISSUE #4953: docs: add detailed pages
## Description
Add detailed documents to describe what is supported, the limitations, how the severity is determined, etc. For example, Trivy prefers Red Hat severities over NVD's severities when scanning Red Hat. Yarn licenses are not yet supported, while npm licenses are supported.

### PR DESCRIPTION
**Pull request by knqyf263**
## Description
Add detailed pages

- Scanner
  - [x] SBOM
  - [x] Vulnerability
- Coverage
  - OS
    - [x] Alpine Linux
    - [x] Wolfi Linux
    - [x] Chainguard
    - [x] RHEL
    - [x] CentOS
    - [x] AlmaLinux
    - [x] RockyLinux
    - [x] Oracle Linux
    - [x] CBL-Mariner
    - [x] Amazon Linux
    - [x] SUSE
    - [x] Photon OS
    - [x] Debian
    - [x] Ubuntu
    - [x] Google Distroless
  - Language
    - [x] Ruby
    - [x] Python
    - [x] PHP
    - [x] Node.js
    - [x] .NET
    - [x] Java
    - [x] Go
    - [x] Rust
    - [x] C/C++
    - [x] Elixir
    - [x] Dart
    - [x] Swift
  - IaC
    - [x] Kubernetes
    - [x] Dockerfile
    - [x] Terraform
    - [x] CloudFormation
    - [x] Azure ARM Template
    - [x] Helm Chart

## Related issues
- Close #4953 
- Close https://github.com/aquasecurity/trivy/issues/4362

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by knqyf263** [comment]
@simar7 @nikpivkin I've also covered IaC files, but I'm sure you're more familiar with them. Please let me know if something is missing or wrong.

@itaysk @AnaisUrlichs Any feedback is welcome.

File: docs/docs/coverage/iac/azure-arm.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+| :----------------: | :-------: |
+| [Misconfiguration] |     ✓     |
+|      [Secret]      |     ✓     |
+
+It supports the following configurations:
+
+|    Format    | Supported |
+| :----------: | :-------: |
+| ARM template |     ✓     |
+|    Bicep     |     ✓     |
```
Comment (nikpivkin) [inline docs/docs/coverage/iac/azure-arm.md:14]: Can we say that Trivy supports `bicep` configuration if the user needs to convert bicep to ARM himself and as a result Trivy will scan ARM?

Comment (knqyf263) [reply docs/docs/coverage/iac/azure-arm.md:14]: I'd say partially. Is there any symbol describing that other than `✓`, `x` or `-`?

File: docs/docs/coverage/iac/docker.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Docker
+Trivy supports the scanners listed in the table below.
+
+|      Scanner       | Supported |
+| :----------------: | :-------: |
+| [Misconfiguration] |     ✓     |
+|      [Secret]      |     ✓     |
+
+It supports the following configigurations.
```
[the remark this answers is not in the document]
Comment (knqyf263) [reply docs/docs/coverage/iac/docker.md]: Fixed in https://github.com/aquasecurity/trivy/pull/4954/commits/b70b2b5c8bbe72879a05bf6a128fedceba626e7e

**Review by DmitriyLewen — APPROVED** [review]

File: docs/docs/coverage/iac/azure-arm.md (hunk 1 above)
Comment (knqyf263) [reply docs/docs/coverage/iac/azure-arm.md]: I've added footnotes. Please let me know if you have any ideas.
https://github.com/aquasecurity/trivy/pull/4954/commits/35e5532f0ce633545c91de00d5b636b630d361fb

**Review by simar7 — APPROVED** [review]

**Review by AnaisUrlichs — APPROVED** [review]

**Pull request merged**